### PR TITLE
Update everything to the latest versions

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: eng/Generate.ps1 [hunk 1]
```diff
@@ -20,8 +20,10 @@ function Invoke-AutoRest($baseOutput, $title, $autoRestArguments)
     }
 
     Write-Host "> $command"
+    pushd $repoRoot
```
Comment (pakrym) [inline eng/Generate.ps1]: had issues with a wrong current directory for cmd invocations

File: eng/Generate.ps1 [hunk 2]
```diff
@@ -5,7 +5,7 @@ $ErrorActionPreference = 'Stop'
 
 function Invoke-AutoRest($baseOutput, $title, $autoRestArguments)
 {
-    $command = "npx autorest-beta $script:debugFlags $autoRestArguments"
+    $command = "npx  @autorest/autorest $script:debugFlags $autoRestArguments"
```
Comment (pakrym) [inline eng/Generate.ps1]: using autorest-beta sometimes calls into global installation make sure we are always calling into local one

File: test/AutoRest.TestServer.Tests/ExtensionsClientNameTest.cs [hunk 3]
```diff
@@ -15,5 +16,12 @@ public void ParametersAreRenamed()
             // TODO: Add more tests here
             TypeAsserts.HasParameter(method, "renamedBodyParameter");
         }
+
+        [Test]
+        public void PropertiesAreRenamed()
+        {
+            TypeAsserts.HasProperty(typeof(OriginalSchema), "RenamedProperty");
+            TypeAsserts.HasProperty(typeof(OriginalSchema), "RenamedPropertyString");
```
Comment (MiYanni) [inline test/AutoRest.TestServer.Tests/ExtensionsClientNameTest.cs]: Is this testing that we (currently) re-case the names of properties? Or is this for future renaming configurability?

File: test/TestServerProjects/extension-client-name/Generated/Models/OriginalSchema.cs [hunk 4]
```diff
@@ -9,6 +9,8 @@ namespace extension_client_name.Models.V100
     public partial class OriginalSchema
     {
         /// <summary> A description about the set of tags. </summary>
-        public IDictionary<string, string>? OriginalProperty { get; set; }
+        public IDictionary<string, string>? RenamedProperty { get; set; }
+        /// <summary> A description about the set of tags. </summary>
+        public string? RenamedPropertyString { get; set; }
```
Comment (MiYanni) [inline test/TestServerProjects/extension-client-name/Generated/Models/OriginalSchema.cs]: What in the autorest update caused this to change? Also, do we need a CodeModel update for the new autorest version?

Comment (pakrym) [reply test/TestServerProjects/extension-client-name/Generated/Models/OriginalSchema.cs]: M4 now picks up x-ms-client-name for more things.

File: test/AutoRest.TestServer.Tests/ExtensionsClientNameTest.cs (hunk 3 above)
Comment (pakrym) [reply test/AutoRest.TestServer.Tests/ExtensionsClientNameTest.cs]: It's testing that m4 applies x-ms-client-name to properties. It didn't before.

**Pull request merged**